internal/internal_task_handlers.go

```diff
@@ -1187,6 +1180,17 @@ func (wth *workflowTaskHandlerImpl) completeWorkflow(
 	}
 }
 
+func errorToFailDecisionTask(taskToken []byte, err error, identity string) *s.RespondDecisionTaskFailedRequest {
+	failedCause := s.DecisionTaskFailedCauseWorkflowWorkerUnhandledFailure
+	_, details := getErrorDetails(err, nil)
```

Comment:
Should we use wth.dataConverter instead of nil here?

Reply:
Ok, then it make sense. 

Would be good to also add a test for this change. 
Other than that LGTM